src/main/java/com/endava/internship/collections/StudentList.java

```diff
@@ -96,59 +93,62 @@ public boolean remove(Object o) {
     @Override
     public void clear() {
         for (int i = 0; i < size; i++) {
-            temp[i] = null;
+            elementArray[i] = null;
         }
         size = 0;
     }
 
     @Override
-    public Student get(int i) {
-        return temp[i];
+    @SuppressWarnings("unchecked")
+    public T get(int i) {
+        return (T) elementArray[i];
     }
 
     @Override
-    public Student set(int i, Student student) {
+    @SuppressWarnings("unchecked")
+    public T set(int i, T t) {
         if (i >= this.size) {
             throw new IndexOutOfBoundsException("Index " + i + " is greater than array size (" + this.size + ")");
         }
-        Student oldValue = temp[i];
-        temp[i] = student;
+        T oldValue = (T) elementArray[i];
+        elementArray[i] = t;
         return oldValue;
     }
 
     @Override
-    public void add(int i, Student student) {
+    public void add(int i, T t) {
         if (size == maxsize) {
             growSize();
         }
         if (size + 1 <= maxsize && i <= size && i >= 0) {
-            for (int j = size - 1; j > i; j--) {
-                temp[j] = temp[j - 1];
+            for (int j = size; j > i; j--) {
+                elementArray[j] = elementArray[j - 1];
             }
-            temp[i] = student;
+            elementArray[i] = t;
             size++;
         }
     }
 
     public void growSize() {
         this.maxsize = maxsize * 2;
-        temp = Arrays.copyOf(temp, maxsize);
+        elementArray = Arrays.copyOf(elementArray, maxsize);
     }
 
     public void growSize(int i) {
         this.maxsize = maxsize + i;
-        temp = Arrays.copyOf(temp, maxsize);
+        elementArray = Arrays.copyOf(elementArray, maxsize);
     }
 
     @Override
-    public Student remove(int i) {
-        temp[i] = null;
-        Student oldValue = null;
+    @SuppressWarnings("unchecked")
+    public T remove(int i) {
```

Comment:
do a check that index index < 0 and index >= length and throw IndexOutOfBoundsException if any of 2 conditions are true

Reply:
Done